hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
+    if [ -z "${vmss_info}" ]; then
+        log "🔴❌🖥️ VMSS '${vmss_name}' in Resource group '$resource_group' does not exist."
+        return 1
+    fi
+
+    # check_jq_installed - Might not needed
+
+    provisioning_state="$( jq -r '.provisioningState' <<< "${vmss_info}")"
+    if [[ "${provisioning_state}" == "Succeeded" ]]; then
+        log "🟢🖥️ VMSS '${vmss_name}' in Resource group '$resource_group' has been provisioned successfully. DELETE_VMSS:${delete_vmss}"
+        if ! is_it_boolean ${delete_vmss} &&  ${delete_vmss}; then
+           az vmss delete --resource-group ${resource_group} --name  ${vmss_name} --force-deletion
+           log "🗑️🖥️ VMSS '${vmss_name}' in Resource group '$resource_group' has been deleted."
+        fi
+    else
+        log "🔴🖥️ VMSS '${vmss_name}' in Resource group '$resource_group' has not been provisioned successfully. Current state: ${provisioning_state}."
+        return 1
+    fi
+}
+
+check_deployment() {
+    err_str="Usage $0 <RESOURCE_GROUP> <DEPLOYMENT_NAME>. Please try again"
+    local resource_group=${1?$err_str}
+    local deployment_name=${2?$err_str}
+
+    # Check if the ResourceGroup exists
+    resource_group_info=$(az group show --resource-group "${resource_group}" 2>/dev/null)
+    if [ -z "${resource_group_info}" ]; then
+        log "🔴❌📦 Resource group '${resource_group}' does not exist."
+        return 1
+    fi
+
+    # Check if the deployment exists
+    deployment_info=$(az deployment group show --resource-group "${resource_group}" --name "${deployment_name}" 2>/dev/null)
+    if [ -z "${deployment_info}" ]; then
+        log "🔴❌📦 Deployment '${deployment_name}' does not exist in resource group '${resource_group}'."
+        return 1
+    fi
+    # Check if the provisioning state is 'Succeeded'
+    # check_jq_installed - Might not needed
+    provisioning_state=$(jq -r '.properties.provisioningState' <<< "${deployment_info}")
+    if [[ "${provisioning_state}" == "Succeeded" ]]; then
+        log "🟢📦 Deployment '${deployment_name}' in resource group '${resource_group}' has been provisioned successfully."
+    else
+        log "🔴📦 Deployment '${deployment_name}' in resource group '${resource_group}' has not been provisioned successfully. Current state: ${provisioning_state}"
+        return 1
+    fi
+}
 
-# Function to extract the image tag - (FUNCTION_NAME)
 extract_image_tag() {
-    # Extract the line containing the return statement
-    local return_line=$(grep -A 1 "func $1" "$2" | grep 'return')
-    echo "$return_line" | sed 's/.*"\(.*\)@sha256.*/\1/'
+    err_str="Usage $0 <IMAGE_NAME> <FILE>. Please try again"
+    local image_name=${1?$err_str}
+    local file=${2?$err_str}
+
+    local return_line
+    return_line=$(awk "/func $image_name/ {flag=1; next} /return/ && flag {print; exit}" "$file")
+    return_line="${return_line#*:}"
+    echo "${return_line%@*}"
+}
+
+VERSION_CONST_FILE="pkg/util/version/const.go"
```

Comment:
This is only referenced by `get_repo_tag()`. Is it intended to be referenced elsewhere in the future?

1. If yes, can we declare it as readonly?
```suggestion
declare -r VERSION_CONST_FILE="pkg/util/version/const.go"
```
2. If no, can we move it into `get_repo_tag()` and conver to lowercase name?

Reply:
2. Done